app.go

```diff
@@ -54,6 +54,36 @@ type AppSettings struct {
 	AsyncModerationConfig     *AsyncModerationConfiguration `json:"async_moderation_config,omitempty"`
 }
 
+type EnforceUniqueUsernames int64
+
+const (
+	No EnforceUniqueUsernames = iota
+	App
+	Team
+)
+
+// check allowed values for enforce_unique_usernames here
+// https://getstream.io/chat/docs/rest/#settings-updateapp
+func (s EnforceUniqueUsernames) String() *string {
+	var result string
+	switch s {
+	case No:
+		result = "no"
+	case App:
+		result = "app"
+	case Team:
+		result = "team"
+	default:
+		result = "app"
+	}
+	return &result
```

Comment:
Does this need to be a pointer? Because I believe that Go's "escape analysis" will put this in the heap instead of stack.

This beauty thing Go does to avoid dangling pointer—since variable should only exists inside the function—always bug my brain. :exploding_head:

Reply:
You're right, it doesn't have to be. Gonna change it later